nemo/core/classes/modelPT.py

```diff
@@ -888,6 +890,38 @@ def set_trainer(self, trainer: 'Trainer'):
         """
         self._trainer = trainer
 
+    def _update_dataset_config(self, dataset_name: str, config: Optional[Union[DictConfig, Dict]]):
+        """
+        Update the config (if not None) of the dataset by given name.
+        Preserves said config after updating.
+
+        Args:
+            dataset_name: str name of the dataset whose config is being updated.
+                Can be one of `train`, `validation` and `test`.
+            config: Optional DictConfig or dict. If None is passed, this method simply returns.
+                If dict is passed, it is cast into a DictConfig.
+                The internal config is updated with the passed config.
+        """
+        if hasattr(self, '_multi_dataset_mode') and self._multi_dataset_mode is True:
+            return
+
+        if config is not None:
+            if not isinstance(config, DictConfig):
+                config = OmegaConf.create(config)
+
+            if dataset_name in ['train', 'validation', 'test']:
+                OmegaConf.set_struct(self.cfg, False)
+
+                key_name = dataset_name + "_ds"
+                self.cfg[key_name] = config
+
+                OmegaConf.set_struct(self.cfg, True)
```

Comment:
Stop using set_struct. Use open_dict() instead

Reply:
Using an additional import for a with context makes no sense imo, when the same functionality exists in a backward compatible way within OmegaConf class itself. Seeing how frequently hydra changes it's api (look at the cls, target then `_targer_`, the removal of params requirement, and finally the deprecation of .pretty() for OmegaConf.to_yaml()) makes me reticent to use anything other than long term apis.